src/fsharp/DependencyManager.Integration.fs

```diff
@@ -0,0 +1,236 @@
+        member __.Key      = instance |> keyProperty
+        member __.ResolveDependencies(targetFramework, scriptDir, mainScriptName, scriptName, packageManagerTextLines) =
+            let arguments = [|box targetFramework; box scriptDir; box mainScriptName; box scriptName; box packageManagerTextLines|]
+            resolveDeps.Invoke(instance, arguments) :?> _
+    interface IDisposable with
+        member __.Dispose () = instance.Dispose()
+            
+
+let assemblySearchPaths = lazy(
+    [ let assemblyLocation =
+          typeof<IDependencyManagerProvider>.GetTypeInfo().Assembly.Location
+      yield Path.GetDirectoryName assemblyLocation
+      let executingAssembly = 
+          typeof<IDependencyManagerProvider>.GetTypeInfo().Assembly
+      yield Path.GetDirectoryName(executingAssembly.Location)
+#if FX_NO_APP_DOMAINS
+#else
+      let baseDir = AppDomain.CurrentDomain.BaseDirectory
+      yield baseDir 
+#endif
+    ]
+    |> List.distinct)
+
+let enumerateDependencyManagerAssembliesFromCurrentAssemblyLocation m =
+    assemblySearchPaths.Force()
+    |> Seq.collect (fun path -> Directory.EnumerateFiles(path,"*DependencyManager*.dll"))
+    |> Seq.choose (fun path -> 
+        try
+            Some(AbstractIL.Internal.Library.Shim.FileSystem.AssemblyLoadFrom path)
+        with 
+        | exn ->
+            warning(Error(FSComp.SR.couldNotLoadDependencyManagerExtenstion(path,exn.Message),m))
+            None)
+    |> Seq.filter (fun a -> ReflectionHelper.assemblyHasAttribute a "FSharpDependencyManagerAttribute")
+
+type ProjectDependencyManager() =
+    interface IDependencyManagerProvider with
+        member __.Name = "Project loader"
+        member __.ToolName = ""
+        member __.Key = "project"
+        member __.ResolveDependencies(_targetFramework:string, _scriptDir: string, _mainScriptName: string, _scriptName: string, _packageManagerTextLines: string seq) = 
+            None,[]
+
+    interface System.IDisposable with
+        member __.Dispose() = ()
+
+type RefDependencyManager() =
+    interface IDependencyManagerProvider with
+        member __.Name = "Ref library loader"
+        member __.ToolName = ""
+        member __.Key = "ref"
+        member __.ResolveDependencies(_targetFramework:string, _scriptDir: string, _mainScriptName: string, _scriptName: string, _packageManagerTextLines: string seq) = 
+            None,[]
+
+    interface System.IDisposable with
+        member __.Dispose() = ()
+
+type ImplDependencyManager() =
+    interface IDependencyManagerProvider with
+        member __.Name = "Impl library loader"
+        member __.ToolName = ""
+        member __.Key = "impl"
+        member __.ResolveDependencies(_targetFramework:string, _scriptDir: string, _mainScriptName: string, _scriptName: string, _packageManagerTextLines: string seq) = 
+            None,[]
+
+    interface System.IDisposable with
+        member __.Dispose() = ()
+
+let registeredDependencyManagers = ref None
```

Comment:
Does this have to be a static global?  Why not hold it in TcConfig or the TcImports (the one for the non-framework assemblies)?